server/src/main/java/org/elasticsearch/index/mapper/InferenceMetadataFieldsMapper.java

```diff
@@ -26,7 +28,37 @@
  * the field name for removal from _source.
  */
 public abstract class InferenceMetadataFieldsMapper extends MetadataFieldMapper {
-    public static final FeatureFlag INFERENCE_METADATA_FIELDS_FEATURE_FLAG = new FeatureFlag("inference_metadata_fields");
+    public static class SystemProperty {
```

Comment:
Wouldn't it be better to add the possibility of using a `FeatureFlag` that is `false` by default, instead of replicating the FF code? 🤔 

(I know this is temporary but just wondering)

Reply:
Yes, but that would require modifying the `FeatureFlag` implementation significantly, and I don't want to open that can of worms right now. This is a temporary workaround to allow us to make this feature opt-in without the possibility of affecting other feature flags.

I plan on starting the discussion about opt-in feature flags in the new year, we can address a long-term solution for situations like this at that time.

Reply:
I understand. I don't like one-off temporary solutions but I get you're trying to get this in for 8.18.

I trust you'll follow up on this.